src/CSModule.sol

```diff
@@ -150,104 +151,53 @@ contract CSModule is
-    ) external payable whenResumed {
-        (uint256 nodeOperatorId, uint256 curveId) = _createNodeOperator(
-            managementProperties,
-            referrer,
-            eaProof
-        );
-
-        if (
-            msg.value < accounting.getBondAmountByKeysCount(keysCount, curveId)
-        ) {
-            revert InvalidAmount();
-        }
-
-        accounting.depositETH{ value: msg.value }(msg.sender, nodeOperatorId);
+    )
+        external
+        onlyRole(CREATE_NODE_OPERATOR_ROLE)
+        whenResumed
+        returns (uint256 nodeOperatorId)
+    {
+        if (from == address(0)) revert ZeroSenderAddress();
 
-        _addKeysAndUpdateDepositableValidatorsCount(
-            nodeOperatorId,
-            keysCount,
-            publicKeys,
-            signatures
-        );
-    }
+        nodeOperatorId = _nodeOperatorsCount;
+        NodeOperator storage no = _nodeOperators[nodeOperatorId];
 
-    /// @inheritdoc ICSModule
-    function addNodeOperatorStETH(
-        uint256 keysCount,
-        bytes calldata publicKeys,
-        bytes calldata signatures,
-        NodeOperatorManagementProperties calldata managementProperties,
-        ICSAccounting.PermitInput calldata permit,
-        bytes32[] calldata eaProof,
-        address referrer
-    ) external whenResumed {
-        (uint256 nodeOperatorId, uint256 curveId) = _createNodeOperator(
-            managementProperties,
-            referrer,
-            eaProof
-        );
+        no.managerAddress = managementProperties.managerAddress == address(0)
+            ? from
+            : managementProperties.managerAddress;
+        no.rewardAddress = managementProperties.rewardAddress == address(0)
+            ? from
+            : managementProperties.rewardAddress;
+        if (managementProperties.extendedManagerPermissions)
+            no.extendedManagerPermissions = true;
 
-        uint256 amount = accounting.getBondAmountByKeysCount(
-            keysCount,
-            curveId
-        );
-        accounting.depositStETH(msg.sender, nodeOperatorId, amount, permit);
+        unchecked {
+            ++_nodeOperatorsCount;
+        }
 
-        _addKeysAndUpdateDepositableValidatorsCount(
+        emit NodeOperatorAdded(
             nodeOperatorId,
-            keysCount,
-            publicKeys,
-            signatures
-        );
-    }
-
-    /// @inheritdoc ICSModule
-    function addNodeOperatorWstETH(
-        uint256 keysCount,
-        bytes calldata publicKeys,
-        bytes calldata signatures,
-        NodeOperatorManagementProperties calldata managementProperties,
-        ICSAccounting.PermitInput calldata permit,
-        bytes32[] calldata eaProof,
-        address referrer
-    ) external whenResumed {
-        (uint256 nodeOperatorId, uint256 curveId) = _createNodeOperator(
-            managementProperties,
-            referrer,
-            eaProof
+            no.managerAddress,
+            no.rewardAddress
```

Comment:
Silly question, maybe, but it looks like we read back from the storage, so can it be better to use  variables on the stack instead?

Reply:
it saves 300 gas